lib/PhasarLLVM/DataFlowSolver/IfdsIde/Problems/IDEGeneralizedLCA/JoinEdgeFunction.cpp

```diff
@@ -37,9 +37,9 @@ JoinEdgeFunction::JoinEdgeFunction(
     auto Ins = Seen.insert(Top);
     if (Top != nullptr && !Ins.second &&
         !dynamic_cast<AllBottom<IDEGeneralizedLCA::l_t> *>(Top)) {
-      std::cerr << "WARNING: cyclic dependency! @" << Ctr << "#";
-      Top->print(std::cerr);
-      std::cerr << std::endl;
+      PHASAR_LOG_LEVEL(WARNING, "cyclic dependency! @" << Ctr << "#");
+      Top->print(llvm::errs());
+      llvm::errs() << '\n';

```

Comment:
Maybe surround with `IF_LOG_ENABLED`

Reply:
I wrote the initial analysis outside of phasar (recall the PG). I don't remember, who ported it to phasar